ndindex/_tuple.pyx

```diff
@@ -51,7 +56,9 @@ cdef object _broadcast_shapes, _BroadcastError
 # We cannot just add these imports to the top because of circular import
 # issues. We can put them inside the constructor, but then they create a big
 # performance bottleneck.
-cdef void _lazy_import():
+# Uses C++ primitives for preventing races
+cdef once_flag _lazy_import_lock
+cdef void _lazy_import() noexcept with gil:
```

Comment:
I've never seen this `with gil` syntax before. Neat!

As far as I can tell it isn't mentioned once in the Cython documentation. I'll open a Cython issue.

Reply:
Oh no I wasn't searching correctly, never mind.